File: apps/api-rs/src/models/identity.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone)]
pub struct CurrentActor {
    pub user_id: Uuid,
    pub tenant_id: Uuid,
    pub email: String,
    pub name: String,
    pub roles: Vec<String>,
    pub permissions: Vec<String>,
    pub allowed_kb_ids: Vec<Uuid>,
    pub is_super_admin: bool,
}

impl CurrentActor {
    pub fn has_role(&self, role: &str) -> bool {
        self.roles.iter().any(|r| r == role)
    }

    pub fn has_permission(&self, permission: &str) -> bool {
        self.permissions.iter().any(|p| p == permission)
    }

    pub fn can_manage_kb(&self, kb_id: Uuid) -> bool {
        self.allowed_kb_ids.contains(&kb_id)
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MeResponse {
    pub user: UserProfile,
    pub tenant: TenantProfile,
    pub roles: Vec<String>,
    pub permissions: Vec<String>,
    pub allowed_kb_ids: Vec<Uuid>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UserProfile {
    pub id: Uuid,
    pub email: String,
    pub name: Option<String>,
    pub avatar_url: Option<String>,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TenantProfile {
    pub id: Uuid,
    pub name: String,
    pub slug: String,
    pub plan: String,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TenantSummary {
    pub id: Uuid,
    pub name: String,
    pub slug: String,
    pub status: String,
    pub plan: String,
    pub member_count: i64,
    pub kb_count: i64,
    pub doc_count: i64,
    pub monthly_queries: i64,
    pub active_admin_count: i64,
    pub pending_invitation_count: i64,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SystemUserSummary {
    pub id: Uuid,
    pub email: String,
    pub name: Option<String>,
    pub status: String,
    pub tenants: Vec<String>,
    pub last_login_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ModelService {
    pub id: Uuid,
    pub name: String,
    pub model: String,
    pub base_url: String,
    pub api_key_tail: String,
    pub status: String,
    pub throughput: String,
    pub latency: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct JobSummary {
    pub id: Uuid,
    pub tenant_id: Uuid,
    pub tenant_name: String,
    pub kind: String,
    pub status: String,
    pub progress: i32,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KnowledgeBaseSummary {
    pub id: Uuid,
    pub tenant_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub status: String,
    pub tags: Vec<String>,
    pub doc_count: i64,
    pub chunk_count: i64,
    pub query_count: i64,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MemberSummary {
    pub id: Uuid,
    pub email: String,
    pub name: Option<String>,
    pub roles: Vec<String>,
    pub allowed_kb_names: Vec<String>,
    pub query_count: i64,
    pub status: String,
    pub joined_at: Option<DateTime<Utc>>,
    pub last_seen_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QaLogSummary {
    pub id: Uuid,
    pub question: String,
    pub kb_name: String,
    pub user_name: String,
    pub score: f64,
    pub feedback: Option<String>,
    pub created_at: DateTime<Utc>,
}

File: apps/api-bun/src/models/identity.ts
// 移植自 apps/api-rs/src/models/identity.rs
export interface CurrentActor {
  user_id: string;
  tenant_id: string;
  login_id: string;
  email: string;
  name: string;
  scope: string;
  roles: string[];
  permissions: string[];
  allowed_kb_ids: string[];
  is_super_admin: boolean;
  api_client_id: string | null;
  api_token_id: string | null;
  api_scopes: string[];
  api_token_expires_at: string | null;
}
export type ActorScope = CurrentActor;

export function actorHasRole(actor: CurrentActor, role: string): boolean {
  return actor.roles.includes(role);
}
export function actorHasPermission(actor: CurrentActor, permission: string): boolean {
  return actor.permissions.includes(permission);
}
export function actorCanManageKb(actor: CurrentActor, kbId: string): boolean {
  return actor.allowed_kb_ids.includes(kbId);
}

export interface UserProfile {
  id: string; login_id: string; email: string; name: string | null;
  avatar_url: string | null; status: string;
}
export interface TenantProfile {
  id: string; name: string; slug: string; plan: string; status: string;
}
export interface MeResponse {
  scope: string; user: UserProfile; tenant: TenantProfile;
  roles: string[]; permissions: string[]; allowed_kb_ids: string[];
}
export interface TenantSummary {
  id: string; name: string; slug: string; status: string; plan: string;
  member_count: number; kb_count: number; doc_count: number; monthly_queries: number;
  active_admin_count: number; pending_invitation_count: number; updated_at: string;
}
export interface SystemUserSummary {
  id: string; login_id: string; email: string; name: string | null; status: string;
  tenants: string[]; last_login_at: string | null;
}
export interface ModelService {
  id: string; name: string; role: string; provider: string; model: string; base_url: string;
  configured: boolean; status: 'healthy' | 'unavailable' | 'disabled';
  latency_ms: number | null; checked_at: string; reason: string | null;
}
export interface JobSummary {
  id: string; tenant_id: string; tenant_name: string; kind: string; title: string;
  status: 'queued' | 'running'; progress: number | null; queue_position: number | null;
  attempt_count: number; max_attempts: number; worker_id: string | null;
  created_at: string; started_at: string | null; updated_at: string;
}
export interface KnowledgeBaseSummary {
  id: string; tenant_id: string; name: string; description: string | null; status: string;
  tags: string[]; doc_count: number; chunk_count: number; query_count: number; updated_at: string;
}
export interface MemberSummary {
  id: string; email: string; name: string | null; roles: string[];
  allowed_kb_names: string[]; query_count: number; status: string;
  joined_at: string | null; last_seen_at: string | null;
}
export interface QaLogSummary {
  id: string; question: string; kb_name: string; user_name: string; score: number;
  feedback: string | null; created_at: string;
}
